### PR TITLE
RTCRtpSender.replaceTrack second cut

### PR DESCRIPTION
**Pull request by jan-ivar**
@stefhak - As you requested, a rebased attempt at capturing or at least getting closer to list consensus, to further discussion. Please comment.

### CODE REVIEW

**Comment by jan-ivar** [comment]
Replaces #195

File: webrtc.html [hunk 1]
```diff
@@ -2537,6 +2537,91 @@
...
+                    treat them as new tracks. The id of the remote track MUST
+                    NOT change in this case.</p>
+                  </li>
+                  <li>
+                    <p>If negotiation is needed, then fire a <a href=
+                      "#event-negotiation">negotiationneeded</a> event at the
+                    <a>RTCPeerConnection</a> associated with this sender.
+                    In this case, <var>p</var> MUST be resolved with
+                    <var>undefined</var> the next time renegotiation is
+                    successfully completed.</p>
```
Comment (stefhak) [inline webrtc.html]: Should there not be text saying that if renegotiation fails, the promise should be rejected? And then the old track should continue flowing (and be returned from sender.track)?
Would we need something similar to the pending SDPs here, a "pending track" that will (or will not) replace the track when negotiation happens?

File: webrtc.html [hunk 2]
```diff
@@ -2537,6 +2537,91 @@
...
+              </li>
+
+              <li>
+                <p>Run the following steps in parallel:</p>
+              </li>
+              <ol>
+                <li>
+                  Set the
+                  <code><a href="#dom-rtpsender-track">RTCRtpSender.track</a></code>
+                  attribute to <var>withTrack</var>.
```
Comment (stefhak) [inline webrtc.html]: Should that not happen once the promise is resolved? Especially if renegotiation is needed, there may be some time before the withTrack is used, and renegotiation may fail.

**Comment by martinthomson** [comment]
From Seattle: we are going to fail if the track can't be seamlessly replaced.

**Comment by martinthomson** [comment]
From Seattle: With the decision to consider Peter's `addMediaSection` proposal, we will have nullable track values on `RTCRtpSender` (and `RTCRtpReceiver`).  I don't think that this needs to change to accommodate that now, but I wanted to record the fact that the argument might need to become nullable too.  This would allow us to model `removeTrack` in terms of `replaceTrack(null)`.

**Comment by jan-ivar** [comment]
Replaced by https://github.com/w3c/webrtc-pc/pull/303.

**Pull request closed without merging**